packages/report-server/src/reportBuilder/types.ts

```diff
@@ -8,3 +8,11 @@ export type FieldValue = string | number | boolean | undefined | null;
 export interface Row {
   [field: string]: FieldValue;
 }
+
+export interface Matrix {
+  columns: {
+    key: string;
+    title: string;
+  }[];
+  rows: Row[];
+}
```

Comment:
Might be nice to define the output types in-line with their output builder functions rather at the top-level here.

We can then export a union type of all possible output types with something like in `outputBuilders.ts`:
```
export type OutputType = ReturnType<typeof outputBuilders[keyof typeof outputBuilders]>;
```

Reply:
Love it! the `ReturnType` is handy and makes things clearer.